src/item.cpp

```diff
@@ -6522,20 +6522,20 @@ void item::mitigate_damage( damage_unit &du ) const
     du.amount = std::max( 0.0f, du.amount );
 }
 
-int item::damage_resist( damage_type dt, bool to_self ) const
+float item::damage_resist( damage_type dt, bool to_self ) const
```

Comment:
Apparently clang does not like that data type change:

```
src/item.cpp:6285:16: error: implicit conversion from 'int' to 'float' changes value from 2147483647 to 2147483648 [-Werror,-Wimplicit-int-float-conversion]

        return INT_MAX;

        ~~~~~~ ^~~~~~~

/usr/lib/llvm-10/lib/clang/10.0.1/include/limits.h:46:19: note: expanded from macro 'INT_MAX'

#define INT_MAX   __INT_MAX__

                  ^~~~~~~~~~~

<built-in>:79:21: note: expanded from here

#define __INT_MAX__ 2147483647

                    ^~~~~~~~~~

src/item.cpp:6319:16: error: implicit conversion from 'int' to 'float' changes value from 2147483647 to 2147483648 [-Werror,-Wimplicit-int-float-conversion]

        return INT_MAX;

        ~~~~~~ ^~~~~~~

/usr/lib/llvm-10/lib/clang/10.0.1/include/limits.h:46:19: note: expanded from macro 'INT_MAX'

#define INT_MAX   __INT_MAX__

                  ^~~~~~~~~~~

<built-in>:79:21: note: expanded from here

#define __INT_MAX__ 2147483647

                    ^~~~~~~~~~

src/item.cpp:6538:30: error: implicit conversion from 'int' to 'float' changes value from 2147483647 to 2147483648 [-Werror,-Wimplicit-int-float-conversion]

            return to_self ? INT_MAX : 0.0f;

            ~~~~~~           ^~~~~~~

/usr/lib/llvm-10/lib/clang/10.0.1/include/limits.h:46:19: note: expanded from macro 'INT_MAX'

#define INT_MAX   __INT_MAX__

                  ^~~~~~~~~~~

<built-in>:79:21: note: expanded from here

#define __INT_MAX__ 2147483647

                    ^~~~~~~~~~

3 errors generated.
```

Reply:
How about replacing `INT_MAX` with `std::numeric_limits<float>::max()`